### PR TITLE
[159003321] Added card validity date check with design…

### PR DESCRIPTION
**Pull request by linsock**
similar to https://projects.invisionapp.com/share/STMZBIQZPB3#/screens
### CARTA SCADUTA
Screenshot 2019-05-08 at 18 33 21

### CARTA VALIDA
Screenshot 2019-05-08 at 18 33 46

There's perhaps the possibility of a better font (style and/or size) with respect to that currently used in the PR, but I think this could be a decision of the design team.

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Please include a Pivotal story at the beginning of the PR title (see below).
  </td>
    </tr>
  </tbody>
</table>



  Example of PR titles that include pivotal stories:

  * single story: `[#123456] my PR title`
  * multiple stories: `[#123456,#123457,#123458] my PR title`

    
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by linsock** [comment]
I've seen my incorrect use of locales.ts which is generated and not mantained, so i'll fix this by editing and committing the make-locales.ts script and reset the locales.ts file.
I've used it for testing purposes but i'll commit a fresh/correct version asap

**Comment by matteodesanti** [comment]
Tks @linsock: may i ask you to add a link in the top-right corner of the card to remove/delete the payment methods?

**Comment by linsock** [comment]
@matteodesanti I fix the locale.ts modify, then I'll add the required link
May I add the requirements in a new pivotal story or do I just simply include in this one?

File: ts/utils/guards.ts [hunk 1]
```diff
@@ -1,6 +1,24 @@
+import { Wallet } from "../types/pagopa";
+
 /**
  * A type guard that filters out undefined and null from a type T
  */
 export function isDefined<T, O extends NonNullable<T>>(v: T): v is O {
   return v !== null && v !== undefined;
 }
+
+export const isExpiredCard = (w: Wallet) => {
```
Comment (fpersico) [inline ts/utils/guards.ts]: Can't be in this file because it is not a guard.

Comment (linsock) [reply ts/utils/guards.ts]: ok, I'll move to a new utility file "checks.ts"

Comment (fpersico) [reply ts/utils/guards.ts]: I think "wallet.ts" is better

**Comment by cloudify** [comment]
This pull request **introduces 1 alert** when merging fac90d0cc87b3ac6d33dd6e7856693631c92a2cb into b3f8204c6ab9d28902c32a93c245b93924892122 - [view on LGTM.com](https://lgtm.com/projects/g/teamdigitale/io-app/rev/pr-06a3ee0be2d615e672ce859634d03107824ac395)

**new alerts:**

* 1 for Invocation of non\-function

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: ts/utils/checks.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { Wallet } from "../types/pagopa";
```
Comment (fpersico) [inline ts/utils/checks.ts]: Rename this file to "wallet.ts" as suggested.

Comment (linsock) [reply ts/utils/checks.ts]: ok

File: ts/utils/wallet.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      .getFullYear()
+      .toString()
+      .slice(2),
+    10
+  );
+  return (
+    parseInt(w.creditCard.expireMonth, 10) > cmpM &&
+    parseInt(w.creditCard.expireYear, 10) <= cmpY
+  );
+};
```
Comment (Undermaken) [inline ts/utils/wallet.ts]: @linsock 
If this function means: "return true when a cc is expired" then the check produces wrong results for certain inputs: i.e 03/19 (return false) or 06/19 (return true)
It should be something:
```javascript 
return (parseInt(w.expireYear, 10) < cmpY ||
parseInt(w.expireYear, 10) == cmpY && parseInt(w.expireMonth, 10) < cmpM));
```
you should write some tests on this check

File: ts/utils/checks.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    const cmpM: number = today.getMonth() + 1;
+    const cmpY: number = parseInt(
+      today
+        .getFullYear()
+        .toString()
+        .slice(2),
+      10
+    );
+    return (
+      parseInt(w.creditCard.expireMonth, 10) > cmpM &&
```
[the remark this answers is not in the document]
Comment (fpersico) [reply ts/utils/checks.ts]: Good catch @Undermaken! Please @linsock change the code and create a test file to check the function logic is correct.

**Pull request merged**